src/pdf_beamtime/include/pdf_beamtime/pdf_beamtime_server.hpp

```diff
@@ -57,9 +59,8 @@ class PdfBeamtimeServer
   std::shared_ptr<rclcpp::ParameterEventHandler> param_subscriber_;
   std::shared_ptr<rclcpp::ParameterCallbackHandle> cb_handle_;
 
-  /// @brief States of the robot transitions
-  enum class State {HOME, PICKUP_APPROACH, PICKUP, GRASP_SUCCESS, PICKUP_RETREAT,
-    PLACE_APPROACH, PLACE, RELEASE_SUCCESS, PLACE_RETREAT};
+  const int num_of_states = 9;
+  InnerStateMachine * state_holder_[9];
```

Comment:
Why do you need 9 distinct Inner state machines?

Reply:
The design of using 9 inner state machines was changed to 1.
https://github.com/maffettone/erobs/blob/c4b4a33b406f989bd7f26a6cec1318fabdf4f4d4/src/pdf_beamtime/src/pdf_beamtime_server.cpp#L53